fmlloader/src/main/java/net/minecraftforge/fml/loading/ModSorter.java

```diff
@@ -235,6 +235,10 @@ private void buildUniqueList()
         LOGGER.debug(LOADING, "Found {} mod requirements missing ({} mandatory, {} optional)", missingVersions.size(), mandatoryMissing, missingVersions.size() - mandatoryMissing);
 
         if (!missingVersions.isEmpty()) {
+            if (mandatoryMissing > 0) {
+                LOGGER.debug(LOADING, "Missing mandatory dependencies: {}", missingVersions.stream().filter(IModInfo.ModVersion::isMandatory).map(IModInfo.ModVersion::getModId).collect(Collectors.joining(", ")));
```

Comment:
Since this information is shown when an error is imminent due to missing mandatory dependencies, a higher logging level would be appropriate here, such as `warn` or `error`.